mmv1/third_party/terraform/resources/resource_storage_transfer_job.go

```diff
@@ -82,6 +82,16 @@ func resourceStorageTransferJob() *schema.Resource {
 					Schema: map[string]*schema.Schema{
 						"object_conditions": objectConditionsSchema(),
 						"transfer_options":  transferOptionsSchema(),
+						"source_agent_pool_name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: `Specifies the agent pool name associated with the posix data source. When unspecified, the default name is used.`,
+						},
+						"sink_agent_pool_name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: `Specifies the agent pool name associated with the posix data source. When unspecified, the default name is used.`,
+						},
```

Comment:
Dose the API return values for these two fields if they are not explicitly specified? (it seems some value like `projects/{projects}}/agentPools/transfer_service_default` is returned based on my local test on `sink_agent_pool_name` and it sounds like it may be the expected behavior according to the field description) If so, we may need to make the field optional + computed to prevent permadiffs in Terraform.

Reply:
Yes it returns the default agent pool name - `projects/{projects}}/agentPools/transfer_service_default`

Reply:
I tested if we can update the fields, but looks like it is not possible, hence adding the forcenew option

Reply:
@shuyama1  I have reached out to the PM of Storage transfer service , This is what i was told.

Two API resource here: AgentPool and TransferJob
> You can update what AgentPool you associate with a Transfer Job by changing fields - sink_agent_pool_name and source_agent_pool_name (that means replace with new agent pool using UpdateTransferJob)
-> AgentPool resource once created doesn't support changing agent name (UpdateAgentPool doesn't support it).

We cannot change the name of the pool name, but a new agentpool can be created and added to a transferjob.

Reply:
Thanks for the update @anuhyapolisetti. It does sound like the update functionality is supported for these two fields in `storage_transfer_job`. We probably don't need forcenew here then. Would you mind adding/modifying an existing test to test the update of these fields (could be from unset of the field (default agent pool) to newly created agent pool). More information on how to add update tests at https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-handwritten-test/#update-tests. (plus, update steps can only be added in handwritten tests, sorry for the extra work!)

Reply:
no worries, Shuya, it's good to have these concerns addressed, my latest commit has the changes for it.